### PR TITLE
[CP-1536] Fix storage memory overflow

### PR DESCRIPTION
**Pull request by pawel-mudita**
The user was able to upload more files than there was free space. In addition, the difference between the declared free space through MTP and MC has been corrected.

**Your checklist for this pull request**

Make sure that this PR:
- [ ] Complies with our guidelines for contributions
- [ ] Has unit tests if possible
- [ ] Has documentation updated
- [ ] Has changelog entry added

### CODE REVIEW

File: module-services/service-desktop/endpoints/filesystem/FS_Helper.cpp [hunk 1]
```diff
@@ -208,6 +209,15 @@ namespace sdesktop::endpoints
             return ResponseContext{.status = code};
         }
 
+        auto freeSpaceLeftForUserFilesMiB = getFreeSpaceForUserFilesMiB();
+        const auto bytesInMebibyte        = 1048576;
```
Comment (dawidwojtas-mudita) [inline module-services/service-desktop/endpoints/filesystem/FS_Helper.cpp:213]: What about `constexpr`? :)

File: module-services/service-desktop/endpoints/filesystem/FS_Helper.cpp [hunk 2]
```diff
@@ -331,4 +341,19 @@ namespace sdesktop::endpoints
...
+    {
+        const auto userDiskPath = purefs::dir::getUserDiskPath();
+
+        std::unique_ptr<struct statvfs> vfstat = std::make_unique<struct statvfs>();
+        if (statvfs(userDiskPath.c_str(), vfstat.get()) < 0) {
+            return 0;
+        }
+
+        unsigned long freeMbytes =
+            (static_cast<uint64_t>(vfstat->f_bfree) * static_cast<uint64_t>(vfstat->f_bsize)) / (1024LLU * 1024LLU) -
```
Comment (dawidwojtas-mudita) [inline module-services/service-desktop/endpoints/filesystem/FS_Helper.cpp:355]: Use `std::uint64_t` if possible.

**Review by dawidwojtas-mudita — APPROVED** [review]

**Review by kkleczkowski — APPROVED** [review]

File: module-services/service-desktop/endpoints/filesystem/FS_Helper.cpp (hunk 1 above)
Comment (pawel-mudita) [reply module-services/service-desktop/endpoints/filesystem/FS_Helper.cpp]: done

File: module-services/service-desktop/endpoints/filesystem/FS_Helper.cpp (hunk 2 above)
Comment (pawel-mudita) [reply module-services/service-desktop/endpoints/filesystem/FS_Helper.cpp]: done

**Pull request merged**